css/cssom-view/scrollLeftTop.html

```diff
@@ -0,0 +1,133 @@
+<!DOCTYPE html>
+<title>CSSOM View - scrollLeft/scrollTop considers writing-mode and css direction</title>
+<meta charset="utf-8">
+<link rel="author" title="Cathie Chen" href="mailto:cathiechen@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#scrolling-area-origin">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#scroll-an-element">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<head>
+  <style>
+    .scroller {
+        overflow: scroll;
+        width: 100px;
+        height: 100px;
+    }
+    .content {
+        width: 300px;
+        height: 300px;
+    }
+    .horizontal-tb {
+      writing-mode: horizontal-tb;
+    }
+    .vertical-lr {
+      writing-mode: vertical-lr;
+    }
+    .vertical-rl {
+      writing-mode: vertical-rl;
+    }
+    .rtl {
+      direction: rtl;
+    }
+  </style>
+</head>
+
+<body>
+  <h1>scrollLeft/scrollTop</h1>
+  <h2>writing-mode: horizontal-tb;</h2>
+  <div id="target_scroller" class="scroller horizontal-tb ltr">
+    <div id="target_content" class="content"></div>
+  </div>
+  <div class="scroller horizontal-tb rtl">
+    <div class="content"></div>
+  </div>
+  <h2>writing-mode: vertical-lr;</h2>
+  <div class="scroller vertical-lr ltr">
+    <div class="content"></div>
+  </div>
+  <div class="scroller vertical-lr rtl">
+    <div class="content"></div>
+  </div>
+  <h2>writing-mode: vertical-rl;</h2>
+  <div class="scroller vertical-rl ltr">
+    <div class="content"></div>
+  </div>
+  <div class="scroller vertical-rl rtl">
+    <div class="content"></div>
+  </div>
+
+  <script>
+  var scroller = document.querySelector("#target_scroller");
+  var content = document.querySelector("#target_content");
+  var scrollbar_width = scroller.offsetWidth - scroller.clientWidth;
+  scroller_width = scroller.offsetWidth;
+  scroller_height = scroller.offsetHeight;
+  content_width = content.offsetWidth;
+  content_height = content.offsetHeight;
+
+  expectedScrollTop = content_height - scroller_height + scrollbar_width;
+  expectedScrollLeft = content_width - scroller_width + scrollbar_width;
+
+  test(() => {
+    var scroller = document.querySelector(".horizontal-tb.ltr");
+    assert_approx_equals(scroller.scrollLeft, 0, 0.5, "initial scrollLeft");
+    assert_approx_equals(scroller.scrollTop, 0, 0.5, "initial scrollTop");
+    scroller.scrollLeft = 1000;
+    scroller.scrollTop = 1000;
```

Comment:
Maybe use 2 * content_width and 2 * content_height to look less arbitrary than 1000.